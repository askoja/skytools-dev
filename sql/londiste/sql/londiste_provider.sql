
set client_min_messages = 'warning';
\set VERBOSITY 'terse'

--
-- tables
--
create table testdata (
    id serial primary key,
    data text
);
create table testdata_nopk (
    id serial,
    data text
);

select current_database();

select * from pgq_set.add_member('aset', 'rnode', 'dbname=db', false);
select * from pgq_set.create_node('aset', 'root', 'rnode', 'londiste_root', null::text, null::int8, null::text);

select * from londiste.node_add_table('aset', 'public.testdata_nopk');
select * from londiste.node_add_table('aset', 'public.testdata');
insert into testdata (data) values ('test-data');
select * from londiste.node_get_table_list('aset');
select * from londiste.node_remove_table('aset', 'public.testdata');
select * from londiste.node_remove_table('aset', 'public.testdata');
select * from londiste.node_get_table_list('aset');

select ev_id, ev_type, ev_data, ev_extra1 from pgq.event_template;

