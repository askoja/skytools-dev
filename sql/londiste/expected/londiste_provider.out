set client_min_messages = 'warning';
\set VERBOSITY 'terse'
--
-- tables
--
create table testdata (
    id serial primary key,
    data text
);
create table testdata_nopk (
    id serial,
    data text
);
select current_database();
 current_database 
------------------
 regression
(1 row)

select * from pgq_set.add_member('aset', 'rnode', 'dbname=db', false);
 ret_code | ret_note 
----------+----------
      200 | Ok
(1 row)

select * from pgq_set.create_node('aset', 'root', 'rnode', 'londiste_root', null::text, null::int8, null::text);
 ret_code | ret_desc 
----------+----------
      200 | Ok
(1 row)

select * from londiste.node_add_table('aset', 'public.testdata_nopk');
 ret_code |                      ret_desc                      
----------+----------------------------------------------------
      400 | Primary key missing on table: public.testdata_nopk
(1 row)

select * from londiste.node_add_table('aset', 'public.testdata');
 ret_code |           ret_desc           
----------+------------------------------
      200 | Table added: public.testdata
(1 row)

select * from londiste.node_get_table_list('aset');
   table_name    | merge_state | custom_snapshot | skip_truncate 
-----------------+-------------+-----------------+---------------
 public.testdata | ok          |                 | 
(1 row)

select * from londiste.node_remove_table('aset', 'public.testdata');
 ret_code |            ret_desc            
----------+--------------------------------
      200 | Table removed: public.testdata
(1 row)

select * from londiste.node_remove_table('aset', 'public.testdata');
 ret_code |          ret_desc          
----------+----------------------------
      400 | Not found: public.testdata
(1 row)

select * from londiste.node_get_table_list('aset');
 table_name | merge_state | custom_snapshot | skip_truncate 
------------+-------------+-----------------+---------------
(0 rows)

